File: sdks/node/examples/agent_with_extractor.ts
import { Agent, Extractor } from 'alith'
import { z } from 'zod'

export const personSchema = z
  .object({
    name: z.string(),
    age: z.number(),
  })
  .strip()

const agent = new Agent({ model: 'gpt-4' })
const extractor = new Extractor(agent, personSchema)
console.log(extractor.extract('Alice is 18 years old!'))

File: sdks/python/examples/agent_with_extractor.py
from alith import Agent, Extractor
from pydantic import BaseModel


class Person(BaseModel):
    name: str
    age: int


print(
    Extractor(
        Agent(
            model="gpt-4",
        ),
        Person,
    ).extract("Alice is 18 years old!")
)
